### PR TITLE
[throttle] Added abstract base class for throttle

### PR DESCRIPTION
**Pull request by abdullahelqaq**
### Summary
This PR adds an [abstract base class](https://www.tutorialspoint.com/cplusplus/cpp_interfaces.htm) (read: skeleton) for the throttle. This class cannot be instantiated as it is, because it has a [pure virtual member function](https://www.tutorialspoint.com/pure-virtual-functions-and-abstract-classes-in-cplusplus). The benefit of having this class exist, however, is that our `motor_control` code can be developed and committed to the codebase with the promise of a certain throttle interface, without the throttle code having to be ready. Throttle implementations can now also be added without having to worry about boilerplate code (an example is included below).

### Test Plan
- [x] This library compiles.

### Example Implementation of Button Throttle
```
class ButtonThrottle: public Throttle {
    public:
        ButtonThrottle(int time_to_full) : Throttle(A0) {/*do stuff*/}
        void compute_motor_value() {/*this is the only function that needs implementing!*/}
};
```

### CODE REVIEW

File: motor_control/throttle/base_throttle.h [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    int get_raw_value() const; // should only be used to debug
+    int get_motor_value() const;
+
+private:
+    int m_raw_value;
+    int m_motor_value;
+    int m_sensor_pin;
+};
+
+#endif
```
Comment (ashvinnagarajan) [inline motor_control/throttle/base_throttle.h]: Such a minor thing. Git has this little warning symbol if you don't have a blank line at the end of a file.

File: motor_control/throttle/base_throttle.h [hunk 2]
```diff
@@ -0,0 +1,21 @@
+#ifndef BASE_THR_H
+#define BASE_THR_H
+
+class Throttle
+{
+public:
+    Throttle(const int &pin);
+
+    virtual void compute_motor_value() = 0; // this needs to be implemented differently for variable/button throttle
```
Comment (ashvinnagarajan) [inline motor_control/throttle/base_throttle.h:9]: Love this so much

**Review by ashvinnagarajan — APPROVED** [review]
Brilliant thinking with this base class @abdullahelqaq! 

I would love to make a similar base class for all of the sensors that are being used on DAQ (IMUs, GPS, hall effect, etc). @anaymehta

File: motor_control/throttle/base_throttle.h (hunk 1 above)
Comment (abdullahelqaq) [reply motor_control/throttle/base_throttle.h]: Lmao will add it.

**Pull request merged**